fastparse/shared/src/test/scala/fastparse/ByteTests.scala

```diff
@@ -266,37 +267,579 @@ object ByteTests extends TestSuite {
       }
     }
     'example{
+      'bs{
+        import fastparse.byte._
+
+        assert(BS(0x01, 0xff).toSeq == Array[Byte](0x01.toByte, 0xff.toByte).toSeq)
+        assert(BS("hello":_*).toSeq == Array[Byte](104, 101, 108, 108, 111).toSeq)
```

Comment:
I made `BS` take anything of type `T: Numeric`; that includes `Byte`, `Short`, `Int` (that's why the `0xff` notation works; it didn't work earlier when it only took `Byte`) and `Char`, which is also defined as `: Numeric`. You can after all add and subtract `Char`s from each other, so in Scala there's an implicit `Numeric[Char]` available. 

Thus, we can take a `String`, splat it into the `BS` function into it's individual chars, and then (at least for ASCII chars) they'll get converted to the correct bytes.

Chars *outside* the byte-range of 0-255 will get truncated to be within 0-255, so all the unicode characters won't work, but ASCII works fine